scipy/sparse/_dok.py

```diff
@@ -344,6 +344,14 @@ def copy(self):
 
     copy.__doc__ = _spbase.copy.__doc__
 
+    @classmethod
+    def fromkeys(cls, iterable, value=1, /):
+        num_rows = 1 + max(k[0] for k in iterable)
+        num_cols = 1 + max(k[1] for k in iterable)
+        result = cls((num_rows, num_cols))
+        result._dict = dict.fromkeys(iterable, value)
```

Comment:
The following change should make this work for the upcoming 1d-arrays with DOK format.
I haven't tested this other than the line to create the `shape` argument.
```suggestion
        d = dict.fromkeys(iterable, value)
        shape = tuple(max(idx) + 1 for idx in zip(*d))
        result = cls(shape)
        result._dict = d
```

Reply:
@dschult  This doen't work for 1d-arrays :

```
>>> tuple(max(idx) + 1 for idx in zip(*d))
Traceback (most recent call last):
  File "/usr/lib/python3.10/idlelib/run.py", line 578, in runcode
    exec(code, self.locals)
  File "<pyshell#51>", line 1, in <module>
TypeError: 'int' object is not iterable
```

Reply:
Sorry about that. The 1D case uses keys as int instead of tuple
```python
        d = dict.fromkeys(iterable, value)
        if isinstance(next(iter(d)), tuple):
            shape = tuple(max(idx) + 1 for idx in zip(*d))
        else:
            shape = (max(d)+1,)
        result = cls(shape)
        result._dict = d
```

Reply:
Okay, I used this approach